cmd/server/pactasrv/initiative.go

```diff
@@ -0,0 +1,164 @@
+	}
+	if b.PactaVersion != nil {
+		mutations = append(mutations, db.SetInitiativePACTAVersion(pacta.PACTAVersionID(*b.PactaVersion)))
+	}
+	if b.PublicDescription != nil {
+		mutations = append(mutations, db.SetInitiativePublicDescription(*b.PublicDescription))
+	}
+	if b.RequiresInvitationToJoin != nil {
+		mutations = append(mutations, db.SetInitiativeRequiresInvitationToJoin(*b.RequiresInvitationToJoin))
+	}
+	err := s.DB.UpdateInitiative(s.DB.NoTxn(ctx), id, mutations...)
+	if err != nil {
+		return errorInternal(err)
+	}
+	return nil
+}
+
+// Deletes an initiative by id
+// (DELETE /initiative/{id})
+func (s *Server) DeleteInitiative(ctx context.Context, request api.DeleteInitiativeRequestObject) (api.DeleteInitiativeResponseObject, error) {
+	err := s.deleteInitiative(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.DeleteInitiativedefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.DeleteInitiative200JSONResponse(emptySuccess), nil
+}
+
+func (s *Server) deleteInitiative(ctx context.Context, request api.DeleteInitiativeRequestObject) error {
+	// TODO(#12) Implement Authorization
+	err := s.DB.DeleteInitiative(s.DB.NoTxn(ctx), pacta.InitiativeID(request.Id))
+	if err != nil {
+		return errorInternal(err)
+	}
+	return nil
+}
+
+// Returns an initiative by ID
+// (GET /initiative/{id})
+func (s *Server) FindInitiativeById(ctx context.Context, request api.FindInitiativeByIdRequestObject) (api.FindInitiativeByIdResponseObject, error) {
+	result, err := s.findInitiativeById(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.FindInitiativeByIddefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.FindInitiativeById200JSONResponse(*result), nil
+}
+
+func (s *Server) findInitiativeById(ctx context.Context, request api.FindInitiativeByIdRequestObject) (*api.Initiative, error) {
+	// TODO(#12) Implement Authorization
+	i, err := s.DB.Initiative(s.DB.NoTxn(ctx), pacta.InitiativeID(request.Id))
+	if err != nil {
+		if db.IsNotFound(err) {
+			return nil, errorNotFound("initiative", request.Id)
+		}
+		return nil, errorInternal(err)
+	}
+	return initiativeToOAPI(i)
+}
+
+// Returns all initiatives
+// (GET /initiatives)
+func (s *Server) ListInitiatives(ctx context.Context, request api.ListInitiativesRequestObject) (api.ListInitiativesResponseObject, error) {
+	result, err := s.listInitiatives(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.ListInitiativesdefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.ListInitiatives200JSONResponse(result), nil
+}
+
+func (s *Server) listInitiatives(ctx context.Context, request api.ListInitiativesRequestObject) ([]api.Initiative, error) {
+	is, err := s.DB.AllInitiatives(s.DB.NoTxn(ctx))
+	if err != nil {
+		return nil, errorInternal(err)
+	}
+	return dereference(mapAll(is, initiativeToOAPI))
```

Comment:
Instead of sprinkling `dereference` everywhere, which makes an additional copy of all the elements, why not just make `initiativeToOAPI` return a non-pointer struct?

Reply:
GQ - this is because In some contexts a nil input to a conv package should yield a nil result, and sometimes it should yield an error. By defaulting to the more lenient version, then making that requirement explicit in a wrapper method, we make the conv package not include bizlogic about which conversions are expected to always be present.